common/src/main/java/com/microsoft/identity/common/internal/cache/CacheKeyValueDelegate.java

```diff
@@ -175,6 +177,14 @@ public String generateCacheKey(Credential credential) {
                 cacheKey = cacheKey.replace(AUTH_SCHEME, sanitizeNull(accessToken.getAccessTokenType()));
             }
 
+            if(!StringExtensions.isNullOrBlank(accessToken.getRequestedClaims()))
+            {
+                // The Requested Claims string has no guarantee it doesn't contain a delimiter, so we hash it
+                cacheKey += CACHE_VALUE_SEPARATOR + REQUESTED_CLAIMS;
+                String reqClaimsHash = String.valueOf(sanitizeNull(accessToken.getRequestedClaims()).hashCode());
```

Comment:
Can't recall off the top of my head: do we need a specific hash alg here? I would think we don't want to rely on Java's built-in `String.hashCode()` and instead want something more x-plat friendly?

Reply:
IMO, HashCode should be fine here. The purpose of adding the claims hash to the cacheKey is to allow 2 "identical" AT's with different claims strings to coexist in the cache. There's no need to be x-plat (Windows and Apple cache implementations both use plat-specific hash algos), so as long as there's a reasonably low chance of collisions (and acceptable perf) I don't think we need to do anything more than HashCode here.

Reply:
ACK